scripts/plotting/tape_recorder.py

```diff
@@ -36,15 +36,27 @@ def tape_recorder(adfobj):
     plot_location = adfobj.plot_location
     plot_loc = Path(plot_location[0])
 
-    #Grab history string:
-    hist_str = adfobj.hist_string
-
     #Grab test case name(s)
     case_names = adfobj.get_cam_info('cam_case_name', required=True)
 
     #Grab test case time series locs(s)
     case_ts_locs = adfobj.get_cam_info("cam_ts_loc", required=True)
 
+    #Grab history strings:
+    cam_hist_strs = adfobj.hist_string["test_hist_str"]
+
+    # Filter the list to include only strings that are exactly in the possible h0 strings
+    # - Search for either h0 of h0a
+    substrings = ["cam.h0","cam.h0a"]
+    case_hist_strs = []
+    for i,cam_case_str in enumerate(cam_hist_strs):
```

Comment:
I believe `substrings` can be a set datatype, also it doesn't look like there is a need to `enumerate`:

```suggestion
    substrings = {"cam.h0","cam.h0a"}
    case_hist_strs = []
    for cam_case_str in cam_hist_strs:
```